framework/decode/vulkan_virtual_swapchain.cpp

```diff
@@ -132,81 +135,153 @@ VkResult VulkanVirtualSwapchain::GetSwapchainImagesKHR(PFN_vkGetSwapchainImagesK
+        {
+            if (props[queue_family_index].queueFlags & VK_QUEUE_GRAPHICS_BIT)
+            {
+                blit_queue_family_index = queue_family_index;
+                device_table_->GetDeviceQueue(device, blit_queue_family_index, 0, &initial_blit_queue);
+                if (initial_blit_queue != VK_NULL_HANDLE)
+                {
+                    break;
+                }
+            }
+        }
+
+        // Make sure we have enough storage for each of our tracked components (Command pools,
+        // Command Buffers, Semaphores, etc) as many queue families that are available.
+        // This is because at any point, the application may get a Device queue from that family and
+        // use it during the present.
+        uint32_t start_size = static_cast<uint32_t>(swapchain_info->blit_command_pools.size());
+        uint32_t new_count  = property_count;
+        if (start_size < new_count)
         {
-            device_table_->GetDeviceQueue(
-                device, swapchain_info->queue_family_indices[0], 0, &swapchain_info->blit_queue);
-
-            VkCommandPoolCreateInfo command_pool_create_info = { VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO };
-            command_pool_create_info.flags =
-                VK_COMMAND_POOL_CREATE_TRANSIENT_BIT | VK_COMMAND_POOL_CREATE_RESET_COMMAND_BUFFER_BIT;
-            command_pool_create_info.queueFamilyIndex = swapchain_info->queue_family_indices[0];
-            VkResult result                           = device_table_->CreateCommandPool(
-                device, &command_pool_create_info, nullptr, &swapchain_info->blit_command_pool);
-            if (result != VK_SUCCESS)
+            swapchain_info->blit_command_pools.resize(new_count);
+            swapchain_info->blit_command_buffers.resize(new_count);
+            swapchain_info->blit_semaphores.resize(new_count);
+            for (uint32_t ii = start_size; ii < new_count; ++ii)
             {
-                return result;
+                swapchain_info->blit_command_pools[ii] = VK_NULL_HANDLE;
             }
 
-            for (uint32_t i = 0; i < capture_image_count; ++i)
+            for (uint32_t queue_family_index = 0; queue_family_index < property_count; ++queue_family_index)
             {
-                VkCommandBufferAllocateInfo allocate_info = { VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO };
-                allocate_info.pNext                       = nullptr;
-                allocate_info.commandPool                 = swapchain_info->blit_command_pool;
-                allocate_info.level                       = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
-                allocate_info.commandBufferCount          = 1;
-                VkCommandBuffer command_buffer            = VK_NULL_HANDLE;
-                result = device_table_->AllocateCommandBuffers(device, &allocate_info, &command_buffer);
-                if (result != VK_SUCCESS)
+                // Create one command pool per queue.
```

Comment:
Maybe we could check if the family index of Queue has VK_QUEUE_GRAPHICS_BIT, and then to Create CommandPool, CommandBuffers and Semaphores